src/main/java/org/smartregister/pathevaluator/action/ActionHelper.java

```diff
@@ -117,14 +117,14 @@ public static ResourceType getResourceType(Action action) {
 
 				return taskDao.findTasksForEntity(entity, planIdentifier);
 			case DEVICE:
-				FHIRPathStringValue locationIdStringValue = PathEvaluatorLibrary.getInstance()
+				FHIRPathStringValue stockIdValue = PathEvaluatorLibrary.getInstance()
 						.evaluateStringExpression(questionnaireResponse,
-								"$this.item.where(linkId='locationId' and definition='details').answer.value.value");
+								"$this.subject.reference.value");
```

Comment:
Why not use `questionnaireResponse.getSubject()`

Reply:
Yes I think we can use `questionnaireResponse.getSubject().getReference().value` directly instead of evaluateStringExpression @githengi

Reply:
yes